perf_timer.py

```diff
@@ -492,6 +504,170 @@ def PrintPerfReport(reportMode, output=None):
 		:type output: None or :class:`collections.Callable` or str
 		"""
 
+		if name is None:
+			name = os.path.basename(sys.modules["__main__"].__file__)
+
+		if output is None:
+			if reportMode != ReportMode.HTML:
+				# pylint: disable=invalid-name,missing-docstring
+				def printIt(*args, **kwargs):
+					print(*args, **kwargs)
+
+				output = printIt
+			else:
+				output = os.path.basename(os.path.splitext(name)[0] + "_PERF.html")
+
+		elementsByFrame = {}
+		earliestByFrame = {}
+		latestByFrame = {}
+		allFramesQueue = deque()
+		while True:
+			try:
+				pair = PerfTimer.perfQueue.popleft()
+				frame = pair[4]
+				elementsByFrame.setdefault(frame, deque()).append(pair)
+				allFramesQueue.append(pair)
+				if frame in earliestByFrame:
+					earliestByFrame[frame] = min(earliestByFrame[frame], pair[5])
+					latestByFrame[frame] = max(latestByFrame[frame], pair[6])
+				else:
+					earliestByFrame[frame] = pair[5]
+					latestByFrame[frame] = pair[6]
+			except IndexError:
+				break
+
+		longest = 0
+		if len(elementsByFrame) != 1 and reportMode == ReportMode.HTML:
```

Comment:
We should really have a check before this `if` statement to check for `elementsByFrame == 0`.  If you attempt to output a report with no perf data, it won't break, but the html output is somewhat invalid.